test/show.jl

```diff
@@ -818,3 +811,20 @@ end
     @test sprint(show, Val(:Float64)) == "Val{:Float64}()" # Val of a symbol
     @test sprint(show, Val(true))     == "Val{true}()"     # Val of a value
 end
+
+@testset "printing of Pair's" begin
+    for (p, s) in (Pair(1.0,2.0)                          => "1.0 => 2.0",
+                   Pair(Pair(1,2), Pair(3,4))             => "(1=>2) => (3=>4)",
+                   Pair{Integer,Int64}(1, 2)              => "Pair{Integer,Int64}(1, 2)",
+                   (Pair{Integer,Int64}(1, 2) => 3)       => "(1=>2) => 3",
```

Comment:
Yes this is loosing info, but I don't mind reverting this; I thougth it made sense to loose info in "compact mode", but it may not actually be used this way. Please, someone make a call on this: revert, use `:limit` property or keep as is.